### ISSUE TITLE
data return 0 rows in match

### ISSUE THREAD

**Issue by wey-gu** [vesoft-inc/nebula#4580]
**Please check the FAQ documentation before raising an issue**

dot involved schema breaks appendVertecies (as row 0)

**Describe the bug (__required__)**

**Your Environments (__required__)**

* OS: `Linux 5.15.0-1014 #17~20.04.1-Ubuntu SMP Thu Jun 23 20:01:51 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* Compiler: `g++ --version` or `clang++ --version`
* CPU: `lscpu`
* Commit id `ef6d6a0` 3.1.0

**How To Reproduce(__required__)**

updated: minimal reproduce as follow

```sql
(root@nebula) [entity_resolution]> :play basketballplayer
Start loading dataset basketballplayer...

Load dataset succeeded!

Wed, 24 Aug 2022 10:18:40 UTC

(root@nebula) [basketballplayer]> match (v:player) return v limit 1
+------------------------------------------------------+
| v                                                    |
+------------------------------------------------------+
| ("player127" :player{age: 42, name: "Vince Carter"}) |
+------------------------------------------------------+
Got 1 rows (time spent 3548/4303 us)

Wed, 24 Aug 2022 10:18:55 UTC

(root@nebula) [basketballplayer]> CREATE TAG `foo.bar` ()
Execution succeeded (time spent 2401/3149 us)

Wed, 24 Aug 2022 10:19:02 UTC

(root@nebula) [basketballplayer]> match (v:player) return v limit 1
+---+
| v |
+---+
+---+
Empty set (time spent 3784/4309 us)

Wed, 24 Aug 2022 10:19:09 UTC

(root@nebula) [basketballplayer]> show hosts meta
+----------+------+----------+--------+--------------+---------+
| Host     | Port | Status   | Role   | Git Info Sha | Version |
+----------+------+----------+--------+--------------+---------+
| "metad1" | 9559 | "ONLINE" | "META" | "33fd35e"    | "3.1.0" |
| "metad0" | 9559 | "ONLINE" | "META" | "33fd35e"    | "3.1.0" |
| "metad2" | 9559 | "ONLINE" | "META" | "33fd35e"    | "3.1.0" |
+----------+------+----------+--------+--------------+---------+
Got 3 rows (time spent 892/1512 us)

```

**Expected behavior**

Data should be consistent to lookup/fetch for MATCH, but got 0 rows after `AppendVertices`
**Additional context**

**Comment by wey-gu** [comment]
reproduce, the long version, for kyle.cao

- deployment

```bash
# deploy nebulagraph with nebula-up (https://github.com/wey-gu/nebula-up/ ) all-in-one mode
curl -fsSL nebula-up.siwei.io/all-in-one.sh | bash

# you could use this instead, as it won't install everything
curl -fsSL nebula-up.siwei.io/all-in-one.sh | bash -s -- v3 dashboard
```
- load data
```bash
~/.nebula-up/console.sh
```
DDL ( I did the equivalent DDL write with studio for the first time)
```sql
CREATE SPACE entity_resolution (vid_type=FIXED_STRING(30));

CREATE TAG `user` (`name` string NOT NULL, `email` string NOT NULL, `phone_num` string NOT NULL, `birthday` date NOT NULL, `address` string NOT NULL)

CREATE TAG `address` ()
CREATE TAG `device` ()
CREATE TAG `email` ()
CREATE TAG `ip` ()
CREATE TAG `phone` ()

CREATE EDGE `used_device` (`time` datetime NOT NULL)
CREATE EDGE `logged_in_from` (`time` datetime NOT NULL)
CREATE EDGE `has_phone` ()
CREATE EDGE `has_address` ()
CREATE EDGE `has_email` ()
```
DML
```sql
INSERT VERTEX `user` (`email`, `name`, `birthday`, `address`, `phone_num`) VALUES
    "user_1":("heathermoore@johnson.com","Miranda Miller",date("1957-08-27"),"Brittany Forge Apt. 718 East Eric  WV 97881","+1-652-450-5443x00562"),
    "user_2":("holly@welch.org","Holly Pollard",date("1990-10-19"),"1 Amanda Freeway Lisaland  NJ 94933","600-192-2985x041"),
    "user_3":("julia.h.24@gmail.com","Julia Hall",date("1927-08-24"),"Rodriguez Track East Connorfort  NC 63144","1248361783"),
    "user_4":("franklin.b@gibson.biz","Franklin Barnett",date("2020-03-01"),"Richard Curve Kingstad  AZ 05660","(224)497-9312"),
    "user_5":("4kelly@yahoo.com","April Kelly",date("1967-12-01"),"Schmidt Key Lake Charles  AL 36174","410.138.1816x98702"),
    "user_6":("steven.web@johnson.com","Steven Webb",date("1955-04-24"),"5 Joanna Key Suite 704 Frankshire  OK 03035","3666519376"),
    "user_7":("Jessica_Torres@morris.com","Jessica Torres",date("1958-09-03"),"1 Payne Circle Mitchellfort  LA 73053","535-357-3112x4903"),
    "user_8":("brettglenn@gmail.com","Brett Glenn",date("1992-09-03"),"Weber Unions Eddieland  MT 64619","660.391.3730"),
    "user_9":("veronica.j@yahoo.com","Veronica Jordan",date("1947-06-08"),"2 Klein Mission New Annetteton  HI 05775","810-252-6218"),
    "user_10":("steven@phelps-craig.info","Steven Brooks",date("1954-06-14"),"1 Vanessa Stravenue Suite 184 Baileyville  NY 46381","+1-665-328-8103x3448"),
    "user_11":("ReginaldTheMan@hotmail.com","Reginald Mccullough",date("1915-04-12"),"John Garden Port John  LA 54602","030.088.4523x94511"),
    "user_12":("Jennifer.f@carroll-acosta.com","Jennifer Foster",date("1988-04-30"),"11 Webb Groves Tiffanyside  MN 14566","(489)306-8558x98227"),
    "user_13":("Philip66@yahoo.com","Philip Garcia",date("1955-12-01"),"70 Robinson Locks Suite 113 East Veronica  ND 87845","490-088-7610x9437"),
    "user_14":("Ann@hernandez.com","Ann Williams",date("1947-05-28"),"24 Mcknight Port Apt. 028 Sarahborough  MD 38195","868.057.4056x4814"),
    "user_15":("Jessica@turner.com","Jessica Stewart",date("1951-11-28"),"0337 Mason Corner Apt. 900 Toddmouth  FL 61464","(335)408-3835x883"),
    "user_16":("Sandra311@hotmail.com","Sandra Dougherty",date("1908-06-03"),"7 Davis Station Apt. 691 Pittmanfort  HI 29746","+1-189-827-0744x27614"),
    "user_17":("Sharon91@gmail.com","Sharon Mccoy",date("1958-09-01"),"1 Southport Street Apt. 098 Westport  KY 85907","(814)898-9079x898"),
    "user_18":("Sharon91+001@gmail.com","Kathryn Miller",date("1958-09-01"),"1 Southport Street Apt. 098 Westport  KY 85907","(814)898-9079x898"),
    "user_19":("brettglenn@googlemail.com","Bretty Glenn",date("1991-09-03"),"Weber Unions Eddieland  MT 64619","660-391-3730"),
    "user_20":("julia.h.24@yahoo.com","Julia H.",date("1927-08-24"),"Rodriguez Track East Connorfort NC 63144","1248361783"),
    "user_21":("holly@welch.org","Holly",date("0000-10-19"),"1 Amanda Freeway Lisaland  NJ 94933","(600)-192-2985"),
    "user_22":("veronica.j@yahoo.com","Veronica Jordan",date("0000-06-08"),"HI 05775","(810)-252-6218"),
    "user_23":("4kelly@hotmail.com","Kelly April",date("2010-01-01"),"Schmidt Key Lake Charles AL 13617","410-138-1816");

INSERT VERTEX `email` () VALUES
    "heathermoore@johnson.com":(),
    "holly@welch.org":(),
    "julia.h.24@gmail.com":(),
    "franklin.b@gibson.biz":(),
    "4kelly@yahoo.com":(),
    "steven.web@johnson.com":(),
    "Jessica_Torres@morris.com":(),
    "brettglenn@gmail.com":(),
    "veronica.j@yahoo.com":(),
    "steven@phelps-craig.info":(),
    "ReginaldTheMan@hotmail.com":(),
    "Jennifer.f@carroll-acosta.com":(),
    "Philip66@yahoo.com":(),
    "Ann@hernandez.com":(),
    "Jessica@turner.com":(),
    "Sandra311@hotmail.com":(),
    "Sharon91@gmail.com":(),
    "Sharon91+001@gmail.com":(),
    "brettglenn@googlemail.com":(),
    "julia.h.24@yahoo.com":(),
    "holly@welch.org":(),
    "veronica.j@yahoo.com":(),
    "4kelly@hotmail.com":();

INSERT EDGE `has_email` () VALUES
    "user_1"->"heathermoore@johnson.com":(),
    "user_2"->"holly@welch.org":(),
    "user_3"->"julia.h.24@gmail.com":(),
    "user_4"->"franklin.b@gibson.biz":(),
    "user_5"->"4kelly@yahoo.com":(),
    "user_6"->"steven.web@johnson.com":(),
    "user_7"->"Jessica_Torres@morris.com":(),
    "user_8"->"brettglenn@gmail.com":(),
    "user_9"->"veronica.j@yahoo.com":(),
    "user_10"->"steven@phelps-craig.info":(),
    "user_11"->"ReginaldTheMan@hotmail.com":(),
    "user_12"->"Jennifer.f@carroll-acosta.com":(),
    "user_13"->"Philip66@yahoo.com":(),
    "user_14"->"Ann@hernandez.com":(),
    "user_15"->"Jessica@turner.com":(),
    "user_16"->"Sandra311@hotmail.com":(),
    "user_17"->"Sharon91@gmail.com":(),
    "user_18"->"Sharon91+001@gmail.com":(),
    "user_19"->"brettglenn@googlemail.com":(),
    "user_20"->"julia.h.24@yahoo.com":(),
    "user_21"->"holly@welch.org":(),
    "user_22"->"veronica.j@yahoo.com":(),
    "user_23"->"4kelly@hotmail.com":();
```

- Then I made queries to see match got 0 rows, and tried created index and lookup to get data properly.

- Then I update the docker-compose yaml file manually to v3.2.0 
```bash
cd ~/.nebula-up/nebula-docker-compose

# change v3.1.0 to v3.2.0 for core images

docker-compose up -d
```

- Things are not changed after upgraded to 3.2.0, same as retry of insert.

**Comment by wey-gu** [comment]
update: it's confirmed it's related the state of the cluster itself, as I cannot reproduce after creating another cluster

**Comment by wey-gu** [comment]
minimal direct cause was narrowed down to this like of ddl in v3.1.0, which is now now allowed in master/v3.2.0

```sql
CREATE TAG `foo.bar` ()
```

**Comment by Sophie-Xie** [comment]
> the minimal direct cause was narrowed down to this dot involved line of ddl in v3.1.0, which is now dis-allowed in master/v3.2.0
> 
> ```sql
> CREATE TAG `foo.bar` ()
> ```
> 
> updated above reproduce procedure

Can we close it？

**Comment by wey-gu** [comment]
> > the minimal direct cause was narrowed down to this dot involved line of ddl in v3.1.0, which is now dis-allowed in master/v3.2.0
> > ```sql
> > CREATE TAG `foo.bar` ()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > updated above reproduce procedure
> 
> Can we close it？

It's not fixed now, this will cause the issue in v3.1.0, but the fix should be cherry-picked to 3.1.0 to be more meaningful.
